src/long-to-linked-pe.cpp

```diff
@@ -0,0 +1,318 @@
+		{ "help", no_argument, &help, 1 },
+		{ "version", no_argument, &version, 1 },
+		{ nullptr, 0, nullptr, 0 }
+	};
+	while ((c = getopt_long(argc, argv, "l:g:o:c:p:sdf:t:b:m:", longopts, &optindex)) != -1) {
+		switch (c) {
+		case 0:
+			break;
+		case 'l':
+			l_set = true;
+			l = std::stoul(optarg);
+			break;
+		case 'm':
+			m = std::stoul(optarg);
+			break;
+		case 'g':
+			g_set = true;
+			g = std::stod(optarg);
+			break;
+		case 'p':
+			dist_read_perc = std::stod(optarg);
+			break;
+		case 'c':
+			cov_to_span = std::stod(optarg);
+			break;
+		case 't':
+			t = std::stoul(optarg);
+		case 'f':
+			configFile = optarg;
+			break;
+		case 'b':
+			bxMultiplicityFile = optarg;
+			break;
+		case 's':
+			auto_span = true;
+			break;
+		case 'd':
+			auto_dist = true;
+			break;
+		default:
+			std::exit(EXIT_FAILURE);
+		}
+	}
+
+	std::vector<std::string> infiles(&argv[optind], &argv[argc]);
+	if (argc < 2) {
+		print_usage();
+		std::exit(EXIT_FAILURE);
+	}
+	if (help != 0) {
+		print_usage();
+		std::exit(EXIT_SUCCESS);
+	} else if (version != 0) {
+		std::cerr << PROGNAME << ' ' << VERSION << std::endl;
+		std::exit(EXIT_SUCCESS);
+	}
+	if (!l_set) {
+		print_error_msg("missing option -- 'l'");
+		failed = true;
+	} else if (l == 0) {
+		print_error_msg("option has incorrect value -- 'l'");
+		failed = true;
+	}
+	if (!g_set) {
+		print_error_msg("missing option -- 'g'");
+		failed = true;
+	} else if (g == 0) {
+		print_error_msg("option has incorrect value -- 'g'");
+		failed = true;
+	}
+	if (infiles.empty()) {
+		print_error_msg("missing file operand");
+		failed = true;
+	}
+	if (failed) {
+		std::cerr << "Try '" << PROGNAME << " --help' for more information.\n";
+		std::exit(EXIT_FAILURE);
+	}
+
+	if (t > MAX_THREADS) {
+		t = MAX_THREADS;
+		std::cerr << (PROGNAME + ' ' + VERSION + ": Using more than " +
+		              std::to_string(MAX_THREADS) + " threads does not scale, reverting to " +
+		              std::to_string(MAX_THREADS) + ".\n")
+		          << std::flush;
+	}
+
+	char header_symbol = '@';
+
+	if (with_fasta) {
+		header_symbol = '>';
+	}
+
+	std::ofstream bx_multiplicity_ofs;
+	if (with_bx_multiplicity_only || with_bx_multiplicity) {
+		bx_multiplicity_ofs = std::ofstream(bxMultiplicityFile, std::ofstream::out);
+	}
+
+	for (auto& infile : infiles) {
+		unsigned flags = 0;
+		flags |= btllib::SeqReader::Flag::NO_FOLD_CASE; // skip time intensive checking
+		flags |= btllib::SeqReader::Flag::NO_TRIM_MASKED;
+		btllib::SeqReader reader(infile, flags, t - 1, 4, 1);
+		btllib::SeqReader::Record record;
+		while ((record = reader.read())) {
+			size_t step = l * 2;
+			std::string& seq = record.seq;
+			size_t seq_size = seq.size();
+			if (step > seq_size || m > seq_size) {
+				continue;
+			}
+			if (with_bx_multiplicity_only || with_bx_multiplicity) {
+				if (seq_size % step != 0) {
+					bx_multiplicity_ofs << record.name << "\t" << seq_size / step + 1 << std::endl;

```

Comment:
This also prints out number of read pairs per barcode, but the ARCS code uses _reads_ per barcode:

```
[lcoombe@hpce704 sam]$ head -n1 test_out.tsv 
bf7b6745-4261-4381-a6d1-1839d21b0884_Basecall_1D_template	64
[lcoombe@hpce704 sam]$ awk '$1==1' ../../barcodeMultiplicityArcs.tsv 
1	128
```

Reply:
updated to reads per barcode.